src/server/api/user.js

```diff
@@ -172,7 +172,7 @@ export const resolvers = {
     },
     assignment: async (user, { campaignId }) =>
       r
-        .reader("assignment")
+        .knex("assignment")
```

Comment:
It's definitely the synchronousness because the reader and primary must have been pointing at the same DB in the environment you were running in!

In GraphQL, you really don't know whether a given resolver is being executed in the context of a query or mutation